net/Socket.hpp

```diff
@@ -409,6 +486,15 @@ class Socket
     std::thread::id _owner;
 };
 
+// injecting specialization of std::hash to namespace std of Socket
+namespace std
+{
+    template <> struct hash<Socket>
```

Comment:
What is the motivation for this? I understand that now we can have `Socket` in unordered_set/map, but we didn't need that so far, so what's the use-case now?

(All our Socket instances are shared_ptr and we intentionally use `vector` for performance. Given our strict ownership model, we cannot have duplicate Sockets. So I'm very curious as to why we need new containers and/or a different model.)

Reply:
It was a WIP snippet for a hash-map, but since I don't need such thing I will remove it.
Yup, as mentioned in the hash snippet, its the unique _fd.

Reply:
removed, thx!